### ISSUE TITLE
error: No such file or directory (os error 2)

### ISSUE THREAD

**Issue by Pipistrel0** [eveningkid/denodb#33]
Hi guys,

### System
Linux Elementary OS 5.1.5
deno 1.0.5
v8 8.4.300
typescript 3.9.2

### Issue
```
import { Database } from "https://deno.land/x/denodb/mod.ts"

export const db = new Database("postgres", {
  username: "bot",
  password: "password",
  database: "bot",
  host: "localhost",
  port: 5435,
})
console.log(db)
```
When I run deno run -A --unstable thiscode.ts it outputs "No such file or directory (os error 2)"

**Comment by maurice-b** [comment]
Hi threre,

### System
Windows 10
Deno 1.0.3
v8: 8.4.300
Typescript: 3.9.2

### Issue
I got the similar issue as Pipistrel0:

```
Debugger listening on ws://127.0.0.1:9229/ws/1710c23a-cfde-46c5-9638-0be58f498dd0
error: The system cannot find the path specified. (os error 3)
thread '<unnamed>' panicked at 'cannot access stderr during shutdown', src\libstd\io\stdio.rs:644
```

I try to debug this issue, but I cannot find anything on the internet what describe: debugging deno or deno libs .
Hopefully you can tell me how you can do that.

**Comment by eveningkid** [comment]
Hello guys, thanks for sharing.

I just upgraded Deno to its last version, and I encourage you to do the same first (just to make sure we're on the same page :)):

```bash
deno upgrade
```

This is my code to test PostgreSQL, and it runs all fine:
```typescript
// postgres.ts

import { DataTypes, Database, Model } from "https://deno.land/x/denodb/mod.ts";

// NOTE: this part works fine on my end
// You can also set debug to true with: new Database({ dialect: "postgres", debug: true }, ...)
const db = new Database("postgres", {
  host: "my_database.azure.com",
  username: "my_username",
  password: "my_password",
  database: "postgres",
});

class Manufacturers extends Model {
  static table = "manufacturers";
  static timestamps = true;
  static fields = {
    id: {
      primaryKey: true,
      autoIncrement: true,
    },
    manufacturer: DataTypes.STRING,
  };
}

class Parts extends Model {
  static table = "parts";
  static timestamps = true;
  static fields = {
    id: {
      primaryKey: true,
      autoIncrement: true,
    },
    part: DataTypes.STRING,
  };
}

db.link([Manufacturers, Parts]);
await db.sync({ drop: true });
await db.close();
```

I then ran the following (just to make sure hehe):

```bash
deno run --allow-net --allow-read --allow-write --unstable --allow-plugin --reload ./postgres.ts
```

If you are still encountering an error, I can only believe this comes from another factor than denoDB itself.

Keep in touch

**Comment by maurice-b** [comment]
I still get this issue.
But.... the issue is also reported by Deno -> https://github.com/denoland/deno/issues/6057

**Comment by eveningkid** [comment]
Thanks for letting me know, I’m sure this will be fixed soon enough.

Let’s wait a little :)

**Comment by 0xdv** [comment]
@eveningkid it still doesn't work for me. do you use 1.0.5 deno version?
could you please share your lock file?

**Comment by eveningkid** [comment]
@0xdv,

It doesn't seem to be related to the lock file. I am running deno 1.0.5 too on macOS 10.15.4.

From the comments I have seen, it seems to happen on both linux and windows, but this is a wild guess it might not be related.

I was still using deno 1.0.0 for a while before upgrading and people didn't seem to have this issue before. **You might want to downgrade for now and see if that fixes the issue.**

In any event, the problem is not related to denodb directly from my understanding. It could be:
- Deno itself
- *Something* used by one of denodb dependencies

We are still waiting on the deno team to get back onto this issue..!

**Comment by 0xdv** [comment]
oh, it seems working with deno 1.0.0. Thanks!

**Comment by yracnet** [comment]
I have the same problem with this environment:
deno --version
deno 1.0.5
v8 8.4.300
typescript 3.9.2

in Linux Mint

**Comment by jsundra** [comment]
Digging into this a little, it doesn't look like it's an issue with denodb, but its dependency on [dex](https://deno.land/x/dex). 

Running `deno info https://deno.land/x/denodb/mod.ts` fails with `error: The system cannot find the path specified. (os error 3)`. Digging through all of the dependencies, they all compile fine except dex. Which gives the same error if you run `deno info https://deno.land/x/dex/mod.ts`.

I'll keep digging there and see if anything comes up.

**Comment by jsundra** [comment]
I got things to work, but I don't have any reason why, other than that the latest version of deno might be doing something weird (an earlier post mentioned deno v1.0.0 working fine).

What I did, was to close denodb & dex locally, and use those as imports. It still failed with using `deno info`. I dug through my local dex clone and just kept brute force commenting out imports and running `deno info` on the imports that seemed problematic. 

I was assuming I'd finally reach an import that was pinned to an older version that needed updating. Nope. When I dug down to some of the single files that were problematic, such as `logger.js` or `transaction.js`, they would run fine with `deno info`, which then when running `deno info mod.ts` on the whole repo, caused that call to succeed.

So with zero local changes to my denodb or dex repos, my project can compile fine (even when running with `--reload`). _shrugs_

**Comment by jsundra** [comment]
Also for what it's worth, I found out you can run `deno info -L debug` to get a ton of data. But it isn't that helpful for debugging, since it will still choke on the path issue with the most recent log entry being the most recent success. But still figured I'd pass on what I learned~

**Comment by yracnet** [comment]
I tried trace the error with the command 
deno info --log-level debug https://deno.land/x/dex/mod.ts
and this returns the error again
"error: No such file or directory (os error 2)"
So I got all the dependences with the command
deno info --log-level debug https://deno.land/x/dex/mod.ts | grep "specifier: https://deno." &> deps.log
with the list I have applied  the command 
deno info --log-level debug ....... 
for each dependent, on my first attempt the error came out, but on the second run this problem was fixed.

**Comment by yracnet** [comment]
I have cleared my cache and run again my script
~~~
deno run --allow-net postgres.ts
~~~
and this returns the error again, then I have run the script dependecies
~~~
deno info --log-level debug ....
~~~
and this problem was fixed.

**Comment by blogwy** [comment]
I have the same problem with this environment:
deno 1.0.5
v8 8.4.300
typescript 3.9.2
 macOS 10.15.5

**Comment by thereis** [comment]
Yup, I'm facing the same issue!

Windows 10
deno 1.0.5
typescript 3.9.2

**Comment by eveningkid** [comment]
Alright guys, we actually have an update on a possible fix:

> bartlomieju: I'm looking into this issue and found root cause. Fix will be released in Deno 1.1 in a few days.

In the meantime, if you want to start using denodb, you should temporarily switch to deno 1.0.0 which worked just fine.

Whenever this is fixed, please someone who first sees can try the new version comment on this thread.

Hopefully, this issue can soon be closed :)

**Comment by linuxing3** [comment]
the same issue!

Windows 10
deno 1.0.5
typescript 3.8,3

**Comment by chrene** [comment]
Not sure if it's related but I get an error when denodb is installed and I have the same issue as everyone else

```
Property 'openPlugin' does not exist on type 'typeof Deno'. 'Deno.openPlugin' is an unstable API. Did you forget to run with the '--unstable' flag?
  return Deno.openPlugin(localPath);
```

**Comment by 0xdv** [comment]
I did `deno run -Ar --unstable app.ts` and got this working on deno 1.0.5. It also installed `deno_mongo` plugin. Why it needs that plugin?

**Comment by eveningkid** [comment]
Alright, they just dropped 1.1.

Run `deno upgrade` and you should be all good.

If it happens not to fix your issue **after upgrading**, feel free to reopen the issue.

**In any event, this issue is not directly related to denodb, but deno itself.**

@0xdv: next time, open another issue as this is not related. About your question, it is just the way `deno_mongo` works internally. As it is part of the dependencies, denodb now requires the `--unstable` option. I will have to see if we can improve this if that bugs some other people.
